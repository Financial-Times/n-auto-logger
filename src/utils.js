export const emptyCheck = value => {
	if (
		value === null ||
		typeof value === 'undefined' ||
		(typeof value === 'string' && value === '')
	) {
		return true;
	}
	return false;
};

export const trimObject = obj => {
	const output = Object.create(obj);
	Object.keys(obj).forEach(key => {
		const emtpy = emptyCheck(obj[key]);
		if (!emtpy) {
			output[key] = obj[key];
		}
	});
	return output;
};

export const removeObjectKeys = obj => keys => {
	if (!Array.isArray(keys)) {
		throw Error('keys need to be formatted in array');
	}
	const output = {};
	Object.keys(obj).forEach(key => {
		const toBeRemoved = keys.includes(key);
		if (!toBeRemoved) {
			output[key] = obj[key];
		}
	});
	return output;
};

export const fieldStringToArray = fieldString =>
	fieldString.split(',').map(keyName => keyName.trim());

export const isPromise = value => Promise.resolve(value) === value;
