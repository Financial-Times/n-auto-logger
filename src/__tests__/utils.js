import { emptyCheck, trimObject, removeObjectKeys } from '../utils';

describe('emptyCheck', () => {
	it('should return true if value is undefined', () => {
		const value = undefined;
		const isEmpty = emptyCheck(value);
		expect(isEmpty).toBe(true);
	});

	it('should return true if value is null', () => {
		const value = null;
		const isEmpty = emptyCheck(value);
		expect(isEmpty).toBe(true);
	});

	it('should return true if value is empty string', () => {
		const value = '';
		const isEmpty = emptyCheck(value);
		expect(isEmpty).toBe(true);
	});

	it('should return false if value is 0', () => {
		const value = 0;
		const isEmpty = emptyCheck(value);
		expect(isEmpty).toBe(false);
	});

	it('should return false if value is boolean false', () => {
		const value = false;
		const isEmpty = emptyCheck(value);
		expect(isEmpty).toBe(false);
	});
});

describe('trimObject', () => {
	it('should remove undefined value', () => {
		const args = { a: undefined, b: 'test' };
		const trimmed = trimObject(args);
		expect(trimmed).toMatchObject({ b: 'test' });
	});

	it('should remove empty string', () => {
		const args = { a: '', b: 'test' };
		const trimmed = trimObject(args);
		expect(trimmed).toMatchObject({ b: 'test' });
	});

	it('should remove null value', () => {
		const args = { a: null, b: 'test' };
		const trimmed = trimObject(args);
		expect(trimmed).toMatchObject({ b: 'test' });
	});

	it('should remove empty values', () => {
		const args = { a: null, b: 'test', c: '', d: undefined };
		const trimmed = trimObject(args);
		expect(trimmed).toMatchObject({ b: 'test' });
	});
});

describe('removeObjectKeys', () => {
	it('remove keys from object', () => {
		const obj = { a: 1, b: 2, c: 'test', 'more-complex': 'test' };
		const removeKeyList = ['a', 'more-complex'];
		const result = removeObjectKeys(obj)(removeKeyList);
		expect(result).toMatchObject({ b: 2, c: 'test' });
	});
});
