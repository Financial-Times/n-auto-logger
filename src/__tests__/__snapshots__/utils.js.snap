// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeObjectKeys throw error if keys input is not an array 1`] = `"keys need to be formatted in array"`;
